plugins/ipam/host-local/backend/allocator/allocator.go

```diff
@@ -215,6 +217,10 @@ func (i *RangeIter) Next() (*net.IPNet, net.IP) {
 
 	if i.startIP == nil {
 		i.startIP = i.cur
+		if nextRange {
+			i.startRange += 1
```

Comment:
hmm, I don't think we should be changing startRange, as we use it to detect if we've looped around.

Reply:
startIP become first ip in next range, but startRange still previous one, so host-local won't jump out from https://github.com/containernetworking/plugins/blob/master/plugins/ipam/host-local/backend/allocator/allocator.go#L91-L104

because `i.rangeIdx == i.startRange && i.cur.Equal(i.startIP)` never true
https://github.com/containernetworking/plugins/blob/master/plugins/ipam/host-local/backend/allocator/allocator.go#L218

Reply:
for example rangeSet: [{start: 10.0.0.1, end: 10.0.0.1}, {start: 10.0.0.3, end: 10.0.0.3}]
it can make startIP 10.0.0.3, startRange 0
however it should be startIP 10.0.0.3, startRange 1

Reply:
> hmm, I don't think we should be changing startRange, as we use it to detect if we've looped around.

@mars1024 I have thought @squeed don't want change startRange in `Next()` function, so I updated this pr.